File: Labs/Lab1/Lab1.py
def twoSum(nums, target):
    map = {}
    n = len(nums)

    for i in range(n): 
        difference = target - nums[i]
        if difference in map:
            return [map[difference], i]
        map[nums[i]] = i
    
    return []
    
arr1 = [2, 7, 11, 15]
print("Test 1:", twoSum(arr1, 9))

arr2 = [3, 2, 4]
print("Test 2:", twoSum(arr2, 6))

arr3 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print("Test 3:", twoSum(arr3, 17))

File: Labs/Lab1/Lab1.java
 import java.util.Arrays;
import java.util.HashMap;

public class Lab1 { 
    public static int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>(); 
        for (int i = 0; i < nums.length; i++) { 
            int difference = target - nums[i];
            if (map.containsKey(difference)) { 
                return new int[]{map.get(difference), i};
            }
            map.put(nums[i], i);
        } 
        return new int[]{};
    }

    
    public static void main(String[] args) {
        int[] arr1 = {2, 7, 11, 15};
        System.out.println("Test 1: " + Arrays.toString(twoSum(arr1, 9)));

        int[] arr2 = {3, 2, 4};
        System.out.println("Test 2: " + Arrays.toString(twoSum(arr2, 6)));

        int[] arr3 = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        System.out.println("Test 3: " + Arrays.toString(twoSum(arr3, 17)));
    }
}